contracts/Minter.sol

```diff
@@ -1,29 +1,62 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.7;
 
-import "hardhat/console.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
 import "./DAvatar.sol";
 
-contract Minter is Ownable {
+contract Minter is Ownable, AccessControlEnumerable {
     uint public fee = 0.01 ether;
```

Comment:
@develo-pera Why public?

Reply:
Because we should be able to fetch it outside the Minter contract

Reply:
Why not make a getter for it? You already have a setter and we should do all settings via this Class API ```setNewFee```, right? What prevents someone from accidentally do uncontrolled change?
```
minterObject.fee = 50000 ether;
```

Reply:
Solidity contracts should not be mistaken for OOP classes, even though there are similarities. In solidity getter is automatically set for a public variable, you don't have to explicitly do it. Thus you can't do `minterObject.fee = 50000 ether;`, but you can do `minter.fee()` and it will return the value of `uint public fee` variable  and `minter.fee` would just return the function itself. Please refer to https://docs.soliditylang.org/en/latest/contracts.html#visibility-and-getters for more information and clarification